inference/huggingface/text-generation/inference-test.py

```diff
@@ -17,14 +17,31 @@
 parser.add_argument("--dtype", default="float16", type=str, choices=["float32", "float16", "int8"], help="data-type")
 parser.add_argument("--ds_inference", action='store_true', help="enable ds-inference")
 parser.add_argument("--use_kernel", action='store_true', help="enable kernel-injection")
+parser.add_argument("--replace_method", required=False, default='', type=str, help="replace method['', 'auto', 'dict']")
 parser.add_argument("--max_tokens", default=1024, type=int, help="maximum tokens used for the text-generation KV-cache")
 parser.add_argument("--max_new_tokens", default=50, type=int, help="maximum new tokens to generate")
 parser.add_argument("--greedy", action='store_true', help="greedy generation mode")
 parser.add_argument("--use_meta_tensor", action='store_true', help="use the meta tensors to initialize model")
 parser.add_argument("--use_cache", default=True, type=bool, help="use cache for generation")
+parser.add_argument("--test_performance", action='store_true', help="enable latency, bandwidth, and throughout testing")
 parser.add_argument("--local_rank", type=int, default=0, help="local rank")
 args = parser.parse_args()
 
+def print_perf_stats(latency_set, config, warmup=3):
+    # trim warmup queries
+    latency_set = list(latency_set)
+    latency_set = latency_set[warmup:]
+    count = len(latency_set)
+
+    if count > 0:
+        latency_set.sort()
+        avg = sum(latency_set) / count
+        num_layers = config.num_layers if hasattr(config,'num_layers') else config.num_hidden_layers
```

Comment:
```suggestion
        num_layers = getattr(config, "num_layers", config.num_hidden_layers)
```

Reply:
We can use the `gettattr` default value (third argument) to make this easier to read